soc/arm/nuvoton_npcx/common/reg/reg_def.h

```diff
@@ -581,12 +581,39 @@ struct adc_reg {
 	volatile uint16_t MEAST;
 };
 
+/* ADC internal inline functions for multi-registers */
 static inline uint32_t npcx_chndat_offset(uint32_t ch)
 {
 	return 0x40 + ch * 2;
 }
 
+static inline uint32_t npcx_thr_base(void)
+{
+	if (IS_ENABLED(CONFIG_SOC_SERIES_NPCX7)) {
+		return 0x014;
+	} else if (IS_ENABLED(CONFIG_SOC_SERIES_NPCX9)) {
+		return 0x060;
+	} else { /* NPCX4 and later series */
+		return 0x080;
+	}
+}
```

Comment:
The following table is the ADC register layout for each npcx series. 

| Regs           | NPCX7   | NPCX9   | NPCX4 |
| :-----         | ----    | ----    | ----  |
| CHNxDAT offset | 0x40 | 0x40 | 0x40 |
| CHNxDAT count  | 10   | 14   | 26   |
| THRCTLx offset | 0x14 | 0x60 | 0x80 |
| THRCTLx count  | 3    | 6    | 6    |
| THEN    offset | X    | X    | 0x90 |
| THEN    count  | X    | X    | 1    |

In most cases,  it is caused by running out of memory space reserved for multi-registers. For these cases, I usually use definitions to wrap them for each series. (Please refer to [lv_gpio_ctl](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nuvoton_npcx/common/reg/reg_def.h#L251C29-L251C47)) We can also put these registers in `adc_reg` structure and use `CONFIG_SOC_SERIES_NPCXX` to distinguish them. But it might get more and more complex if the adc layout is changed in the later series.

Or drop the last commit and use the `threshold-reg-offset` DT prop. instead?

Reply:
I think it's fine to drop the property but I'm just thinking why not group the offset together in defines instead of having an ineline function, like

```c
#ifdef CONFIG_SOC_SERIES_NPCX7
#define NPCX_THR_BASE 0x14
#define NPCX_THEN_OFFSET 0x10
...
#elif CONFIG_SOC_SERIES_NPCX9
...
```

But fine to keep it like this if that's how other npcx drivers has been written already I guess.

Reply:
Yes, you're right. The other npcx drivers follow the same style as this.